### PR TITLE
FIX-#2254: handling dictionary functions at groupby aggregation improved

### LINKED ISSUE #2254: Dictionary-like functions are broken in groupby.aggregation if they passed not by `func` argument
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Any
- **Modin version** (`modin.__version__`): 0.8.1.1
- **Python version**: 3.7.5
- **Code we can use to reproduce**:
```python
if __name__ == "__main__":
    import pandas
    import modin.pandas as pd
    import numpy as np

    data = {
        "col1": [0, 1, 2, 3],
        "col2": [4, 5, 6, 7],
        "col3": [8, 9, 12, 10],
        "col4": [17, 13, 16, 15],
        "col5": [-4, -5, -6, -7],
    }

    pd_grp = pandas.DataFrame(data).groupby("col1")
    md_grp = pd.DataFrame(data).groupby("col1")

    pd_res = pd_grp.agg(max=("col3", np.max))
    md_res = md_grp.agg(max=("col3", np.max))

    print(f"Pandas result:\n{pd_res}")
    print(f"\nModin result:\n{md_res}")

```

<details><summary>Output</summary>

```
Pandas result:
      max
col1
0       8
1       9
2      12
3      10

Modin result:
Empty DataFrame
Columns: []
Index: []

```

</details>

### Describe the problem
The root cause seems to be here
https://github.com/modin-project/modin/blob/dd691cd0d4bb5403b38242563ed8b6ddb316734a/modin/pandas/groupby.py#L303-L305
if we're passing our dict functions via `kwargs` then resulted func-dict will be empty

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2254 
- [x] tests passing

### CODE REVIEW

**Review by prutskov — APPROVED** [review]
Thanks @dchigarev , LGTM!

**Comment by prutskov** [comment]
@devin-petersohn @YarShev Let's merge this if you haven't any comments

File: modin/pandas/groupby.py [hunk 1]
```diff
@@ -301,29 +302,40 @@ def aggregate(self, func=None, *args, **kwargs):
...
         if isinstance(func, dict) or func is None:
-            if func is None:
-                func = {}
-            else:
-                if any(i not in self._df.columns for i in func.keys()):
-                    from pandas.core.base import SpecificationError
 
-                    raise SpecificationError("nested renamer is not supported")
+            def _reconstruct_func(func, **kwargs):
+                relabeling_required, func, new_columns, order = reconstruct_func(
```
Comment (anmyachev) [inline modin/pandas/groupby.py:307]: @dchigarev please describe your solution.

What does the `reconstruct_func` function do?

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: modin/pandas/groupby.py (hunk 1 above)
Comment (dchigarev) [reply modin/pandas/groupby.py:307]: `groupby_dict_agg` query compilers method, assumes that we're passing our dictionary function like:
`{"col_name_to_apply": [func1, func2, ...]}`
hovewer pandas also support cases like this:
`{"new_column_name": ("col_name_to_apply", func)}`
`reconstruct_func` is internal pandas function, that can cast second case to the first + give us meta information about relabeling result.
More in [pandas doc](https://github.com/pandas-dev/pandas/blob/db08276bc116c438d3fdee492026f8223584c477/pandas/core/aggregation.py#L35-L70)

**Review by anmyachev — APPROVED** [review]
LGTM!

**Pull request merged**